connectors/sources/microsoft_teams.py

```diff
@@ -281,6 +720,475 @@ async def ping(self):
             self._logger.exception("Error while connecting to Microsoft Teams")
             raise
 
+    def format_doc(self, item, document_type, **kwargs):
+        document = {}
+        for elasticsearch_field, sharepoint_field in kwargs["document"].items():
+            document[elasticsearch_field] = sharepoint_field
+        self.map_document_with_schema(
+            document=document, item=item, document_type=document_type
+        )
+        return document
+
+    def format_user_chat_meeting_recording(self, item, url):
+        document = {"type": UserEndpointName.MEETING_RECORDING.value}
+        document.update(
+            {
+                "_id": item.get("eventDetail", {}).get("callId"),
+                "title": item.get("eventDetail", {}).get("callRecordingDisplayName"),
+                "url": url,
+                "_timestamp": item.get("lastModifiedDateTime"),
+            }
+        )
+        return document
+
+    def get_calendar_detail(self, calendar):
+        body = ""
+        organizer = calendar.get("organizer", {}).get("emailAddress").get("name")
+        if calendar.get("recurrence"):
+            recurrence_range = calendar.get("recurrence", {}).get("range")
+            pattern = calendar.get("recurrence", {}).get("pattern")
+            occurrence = f"{pattern['interval']}" if pattern.get("interval") else ""
+
+            # In case type of meeting is daily so body will be: Recurrence: Occurs every 1 day starting {startdate}
+            # until {enddate}
+            if pattern.get("type", "") == "daily":
+                days = f"{occurrence} day"
+
+            # If type of meeting  is yearly so body will be: Recurrence: Occurs every year on day 5 of march starting
+            # {date} until {enddate}
+            elif pattern.get("type", "") in ["absoluteYearly", "relativeYearly"]:
+                day_pattern = (
+                    f"on day {pattern['dayOfMonth']}"
+                    if pattern.get("dayOfMonth")
+                    else "on {pattern['index']} {','.join(pattern['daysOfWeek'])}"
+                )
+                days = f"year {day_pattern} of {month_name[pattern['month']]}"
+
+            # If type of meeting  is monthly so body will be: Recurrence: Occurs every month on day 5 of march
+            # starting {date} until {enddate}
+            elif pattern.get("type", "") in ["absoluteMonthly", "relativeMonthly"]:
+                days_pattern = (
+                    f"on day {pattern['dayOfMonth']}"
+                    if pattern.get("dayOfMonth")
+                    else "on {pattern['index']} {','.join(pattern['daysOfWeek'])}"
+                )
+                days = f"{occurrence} month {days_pattern}"
```

Comment:
`pattern.get("type", "")` is called a lot, I think it could be a variable, something like;

```suggestion
            pattern_type = pattern.get("type", "")

            # In case type of meeting is daily so body will be: Recurrence: Occurs every 1 day starting {startdate}
            # until {enddate}
            if pattern_type == "daily":
                days = f"{occurrence} day"

            # If type of meeting  is yearly so body will be: Recurrence: Occurs every year on day 5 of march starting
            # {date} until {enddate}
            elif pattern_type in ["absoluteYearly", "relativeYearly"]:
                day_pattern = (
                    f"on day {pattern['dayOfMonth']}"
                    if pattern.get("dayOfMonth")
                    else "on {pattern['index']} {','.join(pattern['daysOfWeek'])}"
                )
                days = f"year {day_pattern} of {month_name[pattern['month']]}"

            # If type of meeting  is monthly so body will be: Recurrence: Occurs every month on day 5 of march
            # starting {date} until {enddate}
            elif pattern_type in ["absoluteMonthly", "relativeMonthly"]:
                days_pattern = (
                    f"on day {pattern['dayOfMonth']}"
                    if pattern.get("dayOfMonth")
                    else "on {pattern['index']} {','.join(pattern['daysOfWeek'])}"
                )
                days = f"{occurrence} month {days_pattern}"
```